### PR TITLE
AccountRoutingTableDispatcher needs to handle opt-outs.

### PR DESCRIPTION
**Pull request by smn**
Currently that lives in the application message dispatcher, it needs to be ported over.

### CODE REVIEW

File: go/vumitools/routing.py [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+    application_connector_mapping = ConfigDict(
+        "Mapping from conversation_type to connector name.", static=True)
+    message_tag = ConfigList("Tag for the message, if any.")
+    tagpool_metadata = ConfigDict(
+        "Tagpool metadata for the tag attached to the message if any.")
+    conversation_info = ConfigDict(
+        "Conversation info for the tag attached to the message if any.")
+    user_account_key = ConfigText("UserAccount key.")
+    opt_out_connector = ConfigText(
+        "Connector to publish opt-out messages on.", static=True)
```
Comment (hodgestar) [inline go/vumitools/routing.py]: I would move this to the top along with the other static config item.

**Comment by hodgestar** [comment]
:+1: on merging this into #180. The merge will unfortunately be a little messy because some small changes landed in the routing dispatcher (nothing huge, but enough that I think auto merge will fail).

File: go/vumitools/routing.py [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+            msg, self.get_static_config().opt_out_connector, 'default')
+
+    def process_inbound(self, config, msg, connector_name):
+        log.debug("Processing inbound: %s" % (msg,))
+        if not config.message_tag:
+            log.warning("No tag found for inbound message: %s" % (msg,))
+            return
+
+        if OptOutMiddleware.is_optout_message(msg):
+            return self.handle_opt_out(msg)
```
Comment (jerith) [inline go/vumitools/routing.py:89]: I think we still need to add the account key to the helper metadata here.

**Comment by hodgestar** [comment]
Looks good to me.

**Pull request merged**